### DISCUSSION TITLE
How to make a page only available for one locale? (i.e. exclude other locales)

### DISCUSSION THREAD

**Discussion by joelschneider** [amannn/next-intl#955] [Q&A]
Hi all, I have a few pages on my website that are only available in one language (German). Hence, I'd like to disable certain pages, for the other locale (English).

I think my current solution is inelegant (see below).

```
 if (locale === "en") {
    return notFound();
  }
```

Is there a best practice / go-to solution for doing this that I might have overlooked?

Thanks y'all!

```
import React from "react";
import { useTranslations } from "next-intl";
import { unstable_setRequestLocale } from "next-intl/server";
import { locales } from "@/src/config";
import { notFound } from "next/navigation";
// other imports

type Props = {
  params: { locale: string };
};

export default function ExamplePage({ params: { locale } }: Props) {
  // Validate that the incoming `locale` parameter is valid
  const isValidLocale = locales.some((cur) => cur === locale);

  if (!isValidLocale) notFound();

  // Enable static rendering
  unstable_setRequestLocale(locale);

  const t = useTranslations();

  if (locale === "en") {
    return notFound();
  }

  return (
    <div className="grid w-full grid-cols-12 gap-x-6">
      <HeroLeft
        eyebrow={t("ExamplePage.eyebrow")}
        heading={t("ExamplePage.heading")}
        subheader={t("ExamplePage.subheader")}
      />
      <Spacer />
      <TextImage
        imageSrc={"/illustrations/impact-report.webp"}
        imageAlt={t("RolloutGuidePage.1.imageAlt")}
        eyebrow={t("RolloutGuidePage.1.eyebrow")}
        title={t("RolloutGuidePage.1.title")}
        description={t("RolloutGuidePage.1.description")}
        linkHref={t("RolloutGuidePage.downloadLink")}
        linkText={t("RolloutGuidePage.download")}
      />
      <Spacer />
      
      // More code
  );
}
```

**Comment by amannn** [comment]
A similar question is discussed here, in case you want to chime in: https://github.com/amannn/next-intl/discussions/1009

**Reply by joelschneider** [reply]
Thanks for the ping @amannn. Would love to see your suggestion in #1009 realized. Unfortunately, I only started coding 3 months ago and building a multilingual marketing website like Notion.so from the ground up is my first project. Hence, I'm not really qualified to chime in, sorry 🥲 I'm more than happy to elaborate on my use case though (being a technical marketer + building a high-performing localized website for a B2B startup) if that should be of help at some point down the road.

**Reply by amannn** [reply]
> I'm more than happy to elaborate on my use case though (being a technical marketer + building a high-performing localized website for a B2B startup) if that should be of help at some point down the road.

Absolutely, I'm always happy about feedback if you find something that appears to be missing!

**Reply by amannn** [reply]
The issue is being tracked here: https://github.com/amannn/next-intl/issues/990
